### PR TITLE
Add edgex-taf documents

### LINKED ISSUE #13: Add edgex-taf docs
Add documents:
* CH-Edgex-taf.rst for QA tester to implement test cases.
*  CH-Self-Assessment-DS.rst for DS developer to verify DS common behavior

### PR DESCRIPTION
**Pull request by weichou1229**
CH-Edgex-taf.rst for QA tester to implement test cases.
CH-Self-Assessment-DS.rst for DS developer to verify DS common behavior

Fix #13

### CODE REVIEW

**Comment by cloudxxx8** [comment]
@ramya-ranganathan it's the instruction.  could you please review and merge when you have time?  thanks

**Comment by cloudxxx8** [comment]
@ramya-ranganathan , please help take a look.  thank you.

File: docs/CH-Edgex-taf.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+   .. image:: images/Edgex-taf-Robot.png
+       :scale: 50%
+       :alt: Edgex-taf Robot Test Case
+
+6. Put the test scripts in the TAF/testCaseApps folder
+
+7. Run the tests using the following commands::
+
+       python3 edgex-taf-common/TAF-Manager/trigger/run.py -u '*'
```
Comment (ramya-ranganathan) [inline docs/CH-Edgex-taf.rst]: Not clear from which directory this command needs to be given

File: docs/CH-Edgex-taf.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+       docker build -t docker-edgex-taf-common:1.1.1 .
+
+3. Push to the docker registry::
+
+       docker push docker-edgex-taf-common:1.1.1
+
+Run the Tests
+=============
+
+Run the complete test suite as follows::
```
Comment (ramya-ranganathan) [inline docs/CH-Edgex-taf.rst]: Where is the entry point to the TAF ROBOT code when docker run command below is issued.

File: docs/CH-Edgex-taf.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+* Test cases written using human-readable syntax (robotframework does this)
+* Run test scripts using robotframework or dockerized robotframework
+
+  - Deploy test environment ( edgex services for testing)
+  - Run test
+  - Shutdown test environment
+
+* Generate report (robotframework does this)
+* All components are dockerized
+* Integration with Jenkins
```
Comment (ramya-ranganathan) [inline docs/CH-Edgex-taf.rst]: We need a separate PR for the integration to Jenkins.

**Comment by ramya-ranganathan** [comment]
COuld you please add @juan-pablo-lopez-gutierrez and @vishwassp-hcl-intel to the review?

**Comment by cloudxxx8** [comment]
I can't add @juan-pablo-lopez-gutierrez as a reviewer, but please feel free add your review comment here

File: docs/CH-Edgex-taf.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+4. Edit your config and utils under TAF directory:
+
+   .. image:: images/Edgex-taf-TAF.png
+       :scale: 50%
+       :alt: Edgex-taf TAF directory
+
+5. Put the use cases or test cases in the TAF/testScenarios folder
+
+   Create a robot test case must contain the Settings, Variables and Keywords shown below
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Do you mean all keywords need to put in the testcase app folder?
Even the keyword is used for setting up the test suite?
https://github.com/edgexfoundry/edgex-taf/blob/master/TAF/testScenarios/0_deploy/0_deploy.robot

File: docs/CH-Edgex-taf.rst (hunk 2 above)
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: We create the docker image via edgex-taf-common https://github.com/edgexfoundry/edgex-taf-common and you can see the entrypoint file also there.

File: docs/CH-Edgex-taf.rst (hunk 3 above)
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: I create an issue for this task https://github.com/edgexfoundry/edgex-taf/issues/11.

File: docs/CH-Edgex-taf.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
+#########
+EdgeX-taf
+#########
+
+Requirements:
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

File: docs/CH-Edgex-taf.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+* Well-defined project structure
+* Test cases written using human-readable syntax (robotframework does this)
+* Run test scripts using robotframework or dockerized robotframework
+
+  - Deploy test environment ( edgex services for testing)
+  - Run test
+  - Shutdown test environment
+
+* Generate report (robotframework does this)
+* All components are dockerized
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

File: docs/CH-Edgex-taf.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

File: docs/CH-Edgex-taf.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+       git submodule init
+       git submodule update --remote
+
+4. Edit your config and utils under TAF directory:
+
+   .. image:: images/Edgex-taf-TAF.png
+       :scale: 50%
+       :alt: Edgex-taf TAF directory
+
+5. Put the use cases or test cases in the TAF/testScenarios folder
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

File: docs/CH-Edgex-taf.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+   Create a robot test case must contain the Settings, Variables and Keywords shown below
+
+   .. image:: images/Edgex-taf-Robot.png
+       :scale: 50%
+       :alt: Edgex-taf Robot Test Case
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

File: docs/CH-Edgex-taf.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+6. Put the test scripts in the TAF/testCaseApps folder
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: Modified.

**Comment by weichou1229** [comment]
@ramya-ranganathan Changes made, could you help review?

File: docs/CH-Edgex-taf.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (ramya-ranganathan) [reply docs/CH-Edgex-taf.rst]: yes

**Comment by ramya-ranganathan** [comment]
this looks good. Sorry for the late response.

File: docs/CH-Edgex-taf.rst [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+* Edgex-taf-common provides a place for reusable scripts
+
+How to Create an EdgeX-taf Project
+==================================
+
+To write the automation testing with edgex-taf framework, complete the following steps:
+
+1. Clone the edgex-taf project from EdgeX Foundry as a template::
+
+       git clone git@github.com:edgexfoundry/edgex-taf.git
```
Comment (vishwassp-hcl-intel) [inline docs/CH-Edgex-taf.rst]: in edgex-taf I don't see that we have created submodule. Is this intentional ? If not I would prefer to do that 
if we have submodule, we can use --recursive flag to clone the submodule 
git clone --recursive git@github.com:edgexfoundry/edgex-taf.git

Comment (weichou1229) [reply docs/CH-Edgex-taf.rst]: We modify the edgex-taf-common as a python module.
You can install it via pip3
see https://github.com/edgexfoundry/edgex-taf-common/blob/master/README.md

**Comment by weichou1229** [comment]
Merge PR and create a new PR for directory structure change and TAF common usage change.
https://github.com/edgexfoundry/edgex-taf/issues/27
https://github.com/edgexfoundry/edgex-taf-common/blob/master/README.md

**Pull request merged**